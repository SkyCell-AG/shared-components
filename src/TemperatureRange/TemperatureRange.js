import React from 'react'
import PropTypes from 'prop-types'

import Checkbox from '../Checkbox'

const propTypes = {
    showTempRange: PropTypes.bool.isRequired,
    onCheckShowTempRange: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
}

const TemperatureRange = (props) => {
    const {
        showTempRange,
        onCheckShowTempRange,
        title,
    } = props

    return (
        <div>
            <Checkbox
                value={showTempRange}
                onChange={(event) => {
                    onCheckShowTempRange(event)
                }}
                title={title}
            />
        </div>
    )
}

TemperatureRange.propTypes = propTypes

export default TemperatureRange
