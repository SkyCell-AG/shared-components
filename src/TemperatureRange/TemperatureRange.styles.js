import {
    makeStyles,
} from '@material-ui/core'

const useStyles = makeStyles(() => {
    return {
        root: {},
    }
})

export default useStyles
