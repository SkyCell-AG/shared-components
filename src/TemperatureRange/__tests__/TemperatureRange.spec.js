import React from 'react'
import {
    shallow,
} from 'enzyme'

import TemperatureRange from '../TemperatureRange'

describe('shared-components/TemperatureRange', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(
            <TemperatureRange
                showTempRange
                onCheckShowTempRange={jest.fn()}
                title="Show Temperature Range"
            />,
        )

        expect(wrapper).toMatchSnapshot()
    })
})
