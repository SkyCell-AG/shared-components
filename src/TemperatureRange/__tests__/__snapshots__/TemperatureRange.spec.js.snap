// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared-components/TemperatureRange should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <Checkbox
    className=""
    disabled={false}
    indeterminate={false}
    name=""
    onChange={[Function]}
    title="Show Temperature Range"
    value={true}
  />
</div>
`;
